DARNprocessing/utils/utils.py

```
@@ -7,37 +7,89 @@
+
+def flag_options(program_name,program_desc,option_names,option_settings):
+    """
+    Parameter options is a utility to add options to runnable scripts
+
+        :param program_name: name of the program
+        :param program_desc: description of the program
+        :param options: dictionary of dictionaries containing
+                        the following structure:
+                        { '<option name>': {''}
+                            *Note: for required options do not use a hyphen,
+                                   for flag options use a single hyphen for
+                                   singlular letters and two hyphens for words
+        :return options_object: passes back an options object the method can
+                                invoke to obtain the values from sys.argv
+
+    """
+    parser = argparse.ArgumentParser(prog=program_name,
+                                     description=program_desc)
+    # required arguement
+    for option_name, option_setting in zip(option_names,option_settings):
+        if type(option_name) is str:
+            parser.add_argument(option_name,**option_setting)
+        else:
+            parser.add_argument(*option_name,**option_setting)
+
+
+    parameter = parser.parse_args()
+    return vars(parameter)
+
+def path_exists(path):
+    """
+    Check if a path exists
+
+        :param path: path to check
+        :raise PathDoesNotExistException: path does not exist
+    """
+    if not os.path.exists(path):
+        raise PathDoesNotExistException(path)
+
 
 def file_exists(filename):
     """
     Checks if a file exists if not raises an IOError
+
+        :param filename: name of the file name to check
+        :return: boolean true if file exists
+        :raises IOError: if file does not exist
     """
     if not os.path.isfile(filename):
         message = "{} does not exists".format(filename)
         raise IOError(message)
```

I think there are two options for this method: change to just return True or False, or change the name to except_on_no_file (for example) and let it raise the error (and you can remove the return True).